Src/FluentAssertions/Events/EventMonitor.cs

```diff
@@ -117,20 +121,48 @@ public void Dispose()
     {
         foreach (EventRecorder recorder in recorderMap.Values)
         {
-            recorder.Dispose();
+            DisposeSafeIfRequested(recorder);
         }
 
         recorderMap.Clear();
     }
 
+    private void DisposeSafeIfRequested(EventRecorder recorder)
+    {
+        try
+        {
+            recorder.Dispose();
+        }
+        catch when (options.ShouldIgnoreEventAccessorExceptions)
+        {
+            // ignore
+        }
+    }
+
     private void AttachEventHandler(EventInfo eventInfo, Func<DateTime> utcNow)
     {
         if (!recorderMap.TryGetValue(eventInfo.Name, out _))
         {
             var recorder = new EventRecorder(subject.Target, eventInfo.Name, utcNow, threadSafeSequenceGenerator);
             if (recorderMap.TryAdd(eventInfo.Name, recorder))
             {
-                recorder.Attach(subject, eventInfo);
+                AttachEventHandler(eventInfo, recorder);
+            }
+        }
+    }
+
+    private void AttachEventHandler(EventInfo eventInfo, EventRecorder recorder)
+    {
+        try
+        {
+            recorder.Attach(subject, eventInfo);
+        }
+        catch when (options.ShouldIgnoreEventAccessorExceptions)
+        {
+            if (!recorderMap.TryRemove(eventInfo.Name, out _))
```

Comment:
🔧 Instead of removing the event from the map at this point (which is what I think caused the confusion), have this method return a `bool`. And then in caller, do something like 

```csharp
            if (recorderMap.TryAdd(eventInfo.Name, recorder))
            {
                if (!AttachEventHandler(eventInfo, recorder))
                {
                    recorderMap.TryRemove(eventInfo.Name, recorder);
                }
            }
```

I don't think we should be throwing exceptions. Somebody _could_ have disposed the monitor.

Reply:
Hi, sorry for the long delay. I checked out the code and when you call the AttachEventHandler with the EventRecorder argument and this calles tryremove anyway, then the second try remove does not do anything, as far as I understand. Because you are calling the tryremove two times in a row, it returns the true and then you call the try again.

The use case here is: Ignore the error when a custom event accessor throws any exception on attaching. When it is not on the recorder map and thus not removed from it, then it is fine.

TLDR: I just skipped throwing the exception when the EventRecorder was not in the recorderMap dictionary, or it was removed twice, because I don't think it matters anyway for the tests.